subsys/bluetooth/mesh/cfg.c

```diff
@@ -0,0 +1,228 @@
+	} else {
+		bt_mesh_beacon_disable();
+	}
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS) &&
+	    atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
+		bt_mesh_store_cfg();
+	}
+}
+
+enum bt_mesh_feat_state bt_mesh_beacon_get(void)
+{
+	return atomic_test_bit(bt_mesh.flags, BT_MESH_BEACON);
+}
+
+int bt_mesh_gatt_proxy_set(bool gatt_proxy)
+{
+	if (!IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY)) {
+		return -ENOTSUP;
+	}
+
+	atomic_set_bit_to(bt_mesh.flags, BT_MESH_GATT_PROXY, gatt_proxy);
+
+	bt_mesh_hb_feature_changed(BT_MESH_FEAT_PROXY);
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS) &&
+	    atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
+		bt_mesh_store_cfg();
+	}
+
+	return 0;
+}
+
+enum bt_mesh_feat_state bt_mesh_gatt_proxy_get(void)
+{
+	if (!IS_ENABLED(CONFIG_BT_MESH_GATT_PROXY)) {
+		return BT_MESH_FEATURE_NOT_SUPPORTED;
+	}
+
+	return atomic_test_bit(bt_mesh.flags, BT_MESH_GATT_PROXY);
+}
+
+int bt_mesh_default_ttl_set(uint8_t default_ttl)
+{
+	if (default_ttl == 1 || default_ttl > BT_MESH_TTL_MAX) {
+		return -EINVAL;
+	}
+
+	if (default_ttl == bt_mesh.default_ttl) {
+		return 0;
+	}
+
+	bt_mesh.default_ttl = default_ttl;
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS) &&
+	    atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
+		bt_mesh_store_cfg();
+	}
+
+	return 0;
+}
+
+uint8_t bt_mesh_default_ttl_get(void)
+{
+	return bt_mesh.default_ttl;
+}
+
+int bt_mesh_friend_set(bool friendship)
+{
+	if (!IS_ENABLED(CONFIG_BT_MESH_FRIEND)) {
+		return -ENOTSUP;
+	}
+
+	if (atomic_test_bit(bt_mesh.flags, BT_MESH_FRIEND) == friendship) {
+		return 0;
+	}
+
+	atomic_set_bit_to(bt_mesh.flags, BT_MESH_FRIEND, friendship);
+
+	bt_mesh_hb_feature_changed(BT_MESH_FEAT_FRIEND);
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS) &&
+	    atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
+		bt_mesh_store_cfg();
+	}
+
+	if (!friendship) {
+		bt_mesh_friends_clear();
+	}
+
+	return 0;
+}
+
+enum bt_mesh_feat_state bt_mesh_friend_get(void)
+{
+	if (!IS_ENABLED(CONFIG_BT_MESH_FRIEND)) {
+		return BT_MESH_FEATURE_NOT_SUPPORTED;
+	}
+
+	return atomic_test_bit(bt_mesh.flags, BT_MESH_FRIEND);
+}
+
+void bt_mesh_net_transmit_set(uint8_t xmit)
+{
+	if (bt_mesh.net_xmit == xmit) {
+		return;
+	}
+
+	bt_mesh.net_xmit = xmit;
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS) &&
+	    atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
+		bt_mesh_store_cfg();
+	}
+}
+
+uint8_t bt_mesh_net_transmit_get(void)
+{
+	return bt_mesh.net_xmit;
+}
+
+int bt_mesh_relay_set(bool relay, uint8_t xmit)
+{
+	if (!IS_ENABLED(CONFIG_BT_MESH_RELAY)) {
+		return -ENOTSUP;
+	}
+
+	bt_mesh.relay_xmit = xmit;
+
+	if (relay != atomic_test_bit(bt_mesh.flags, BT_MESH_RELAY)) {
+		atomic_set_bit_to(bt_mesh.flags, BT_MESH_RELAY, relay);
+		bt_mesh_hb_feature_changed(BT_MESH_FEAT_RELAY);
+	}
+
+	if (IS_ENABLED(CONFIG_BT_SETTINGS) &&
+	    atomic_test_bit(bt_mesh.flags, BT_MESH_VALID)) {
+		bt_mesh_store_cfg();
+	}
+
+	return 0;
+}
+
+enum bt_mesh_feat_state bt_mesh_relay_get(void)
+{
+	return atomic_test_bit(bt_mesh.flags, BT_MESH_RELAY);
```

Comment:
Return value of atomic_test_bit is bool, it looks wrong to have enum bt_mesh_feat_state as return value.

Reply:
This is a similar problem to the set API bool, I'll convert it.